### DISCUSSION TITLE
Is there a way to get the total count of tests executed by SBT?

### DISCUSSION THREAD

**Discussion by yadavan88** [sbt/sbt#7428] [Q&A]
I am managing a huge Scala tutorial repo and I want to get the total number of tests executed by SBT. Since it is a tutorial repo, it contains tests using all testing framework. 
At the moment, I used the jenkins log manipulation with basic string operations to find the count. 

Since SBT already logs the info like below for each sub-module:
`Tests: succeeded 14, failed 0, canceled 0, ignored 0, pending 0`

I was wondering if there is already some way to get the total number of tests executed by SBT across all sub modules?

Thanks in advance
Yadu

**Comment by eed3si9n** [comment]
You can do something like this:

```scala
lazy val testCount = taskKey[Int]("test count")
val inAllTest = ScopeFilter(inAnyProject, inConfigurations(Test))
testCount := {
  val xss: Seq[Seq[String]] = definedTestNames.all(inAllTest).value
  xss.flatten.size
}
```

```scala
> show testCount
[info] 73
```

**Reply by yadavan88** [reply]
Hi @eed3si9n 
Thanks for the quick reply. This is working, but I see a large difference in the count here and the number of tests executed in jenkins. I am only comparing the tests in the Test scope. Just in case you are wondering which repo I am talking about, it is the [Baeldung](https://github.com/Baeldung/scala-tutorials) scala tutorial repository.

**Reply by eed3si9n** [reply]
You can try adding `inConfigurations(Test, IntegrationTest)` in there.

**Reply by yadavan88** [reply]
I am only executing unit tests in the jenkins. There I can see totally 1116 tests are expected. But this task is showing the count around 300. I am wondering what is missing there

**Reply by yadavan88** [reply]
Ah, OK. I think got what is wrong. it shows the number of test classes right? This is also good.
I was looking for the actual number of test cases. Is it anyways possible? 
Thanks
